src/XR/webXRSessionManager.ts

```diff
@@ -190,6 +191,11 @@ export class WebXRSessionManager implements IDisposable, IWebXRRenderTargetTextu
         if (!this._xrNavigator.xr) {
             return Promise.reject("WebXR not available");
         }
+
+        if (this.isNative) {
+            NativeXRPlugin.ExtendNativeObject();
+        }

```

Comment:
So is the pattern here that code can call `RegisterType` early (before `_native` exists in the global scope), and then by the time this `ExtendNativeObject` function is called `_native` will exist? If so, I wonder if instead it might make sense to more generically have JS side initialization code that either executes synchronously if `_native` already exists, and if not have it use `Object.defineProperty` in the global scope to define a `_native` property such that when the native side sets that global property, we could have a callback (as part of the `Object.defineProperty`) that could do the deferred initialization. It seems like that would be a generic enough mechanism to support most if not all of our JS side initialization upon `_native` being defined. Thoughts?

Reply:
I changed the functions a bit to set the properties on _native directly if _native has already been defined, otherwise they're saved for later.
The `Object.defineProperty` solution is interesting, I just hesitate to define `_native` behind a setter/getter for performance reasons, and it feels like it warrants discussion as to whether we create an observable so that we could define multiple behaviors on definition of _native. If it's a blocker I can think about it some more